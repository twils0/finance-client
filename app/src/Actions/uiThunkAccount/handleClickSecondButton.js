import { codeTypeNames } from '../../Constants/dataConstantsAuth';
import { formNames } from '../../Constants/uiConstantsAccount';
import requestVerifyField from '../dataThunkAuth/requestVerifyField';
import handleCancel from '../uiThunkAccount/handleCancel';

const handleClickSecondButton = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'resetStripeElement')) {
    throw new Error(
      `Please enter a value for the 'resetStripeElement' key - ${JSON.stringify(payload)}`,
    );
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.internal.account;
    const { codeTypes } = state.data.auth;

    switch (forms.current) {
      case formNames.CODE: {
        if (codeTypes[codeTypeNames.VERIFY_PHONE].needed) {
          dispatch(requestVerifyField({ field: 'phone_number' }));
        }
        break;
      }
      case formNames.BILLING: {
        payload.resetStripeElement();
        break;
      }
      default:
        break;
    }

    dispatch(handleCancel());

    return null;
  };
};

export default handleClickSecondButton;
