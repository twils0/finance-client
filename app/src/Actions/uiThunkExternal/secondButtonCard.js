import { pathNames } from '../../Constants/universalConstants';
import { formNames, inputNames } from '../../Constants/uiConstantsExternal';
import { setInputValueError } from '../uiActionsExternal';

const secondButtonCard = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'history')) {
    throw new Error(`Please enter a value for the 'history' key - ${JSON.stringify(payload)}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'resetStripeElement')) {
    throw new Error(`Please enter a value for the 'resetStripeElement' key - ${JSON.stringify(payload)}`);
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.external;
    const { history } = payload;

    const cardNameOnCard = forms[formNames.CARD].inputs[inputNames[formNames.CARD].NAME_ON_CARD];
    const cardPromoCode = forms[formNames.CARD].inputs[inputNames[formNames.CARD].PROMO_CODE];
    const cardStripe = forms[formNames.CARD].inputs[inputNames[formNames.CARD].STRIPE];

    const infoName = forms[formNames.INFO].inputs[inputNames[formNames.INFO].NAME];
    const infoEmail = forms[formNames.INFO].inputs[inputNames[formNames.INFO].EMAIL];
    const infoPhone = forms[formNames.INFO].inputs[inputNames[formNames.INFO].PHONE];

    const signUpPassword = forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD];
    const signUpPassword2 =
      forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD2];

    history.replace(pathNames.LOGIN);

    if (cardNameOnCard.value || cardNameOnCard.errorMessage) {
      cardNameOnCard.value = '';
      cardNameOnCard.errorMessage = '';

      dispatch(setInputValueError(cardNameOnCard));
    }

    if (cardPromoCode.value || cardPromoCode.errorMessage) {
      cardPromoCode.value = '';
      cardPromoCode.errorMessage = '';

      dispatch(setInputValueError(cardPromoCode));
    }

    if (cardStripe.value || cardStripe.errorMessage) {
      cardStripe.value = '';
      cardStripe.errorMessage = '';

      dispatch(setInputValueError(cardStripe));
      payload.resetStripeElement();
    }

    if (infoName.value || infoName.errorMessage) {
      infoName.value = '';
      infoName.errorMessage = '';

      dispatch(setInputValueError(infoName));
    }

    if (infoEmail.value || infoEmail.errorMessage) {
      infoEmail.value = '';
      infoEmail.errorMessage = '';

      dispatch(setInputValueError(infoEmail));
    }

    if (infoPhone.value || infoPhone.errorMessage) {
      infoPhone.value = '';
      infoPhone.errorMessage = '';

      dispatch(setInputValueError(infoPhone));
    }

    if (signUpPassword.value || signUpPassword.errorMessage) {
      signUpPassword.value = '';
      signUpPassword.errorMessage = '';

      dispatch(setInputValueError(signUpPassword));
    }

    if (signUpPassword2.value || signUpPassword2.errorMessage) {
      signUpPassword2.value = '';
      signUpPassword2.errorMessage = '';

      dispatch(setInputValueError(signUpPassword2));
    }

    return null;
  };
};

export default secondButtonCard;
