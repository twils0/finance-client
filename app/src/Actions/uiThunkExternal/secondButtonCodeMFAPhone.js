import { pathNames, requestStatusTypes } from '../../Constants/universalConstants';
import { statusNames } from '../../Constants/dataConstantsAuth';
import { formNames, inputNames } from '../../Constants/uiConstantsExternal';
import { setAuthStatus } from '../dataActionsAuth';
import { setInputValueError } from '../uiActionsExternal';

const secondButtonCodePhone = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'history')) {
    throw new Error(`Please enter a value for the 'history' key - ${JSON.stringify(payload)}`);
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.external;
    const { status } = state.data.auth;

    const codeCodeMFAPhone =
      forms[formNames.CODE_MFA_PHONE].inputs[inputNames[formNames.CODE_MFA_PHONE].CODE];

    const loginEmail = forms[formNames.LOGIN].inputs[inputNames[formNames.LOGIN].EMAIL];
    const loginPassword = forms[formNames.LOGIN].inputs[inputNames[formNames.LOGIN].PASSWORD];

    payload.history.replace(pathNames.LOGIN);

    if (status[statusNames.SIGN_UP].status !== requestStatusTypes.IDLE) {
      dispatch(setAuthStatus({
        id: statusNames.SIGN_UP,
        status: requestStatusTypes.IDLE,
      }));
    }

    if (status[statusNames.LOGIN].status !== requestStatusTypes.IDLE) {
      dispatch(setAuthStatus({
        id: statusNames.LOGIN,
        status: requestStatusTypes.IDLE,
      }));
    }

    if (status[statusNames.LOGIN_MFA].status !== requestStatusTypes.IDLE) {
      dispatch(setAuthStatus({
        id: statusNames.LOGIN_MFA,
        status: requestStatusTypes.IDLE,
      }));
    }

    if (status[statusNames.SIGN_OUT_DEVICES].status !== requestStatusTypes.IDLE) {
      dispatch(setAuthStatus({
        id: statusNames.SIGN_OUT_DEVICES,
        status: requestStatusTypes.IDLE,
      }));
    }

    if (codeCodeMFAPhone.value || codeCodeMFAPhone.errorMessage) {
      codeCodeMFAPhone.value = '';
      codeCodeMFAPhone.errorMessage = '';

      dispatch(setInputValueError(codeCodeMFAPhone));
    }

    if (loginEmail.value || loginEmail.errorMessage) {
      loginEmail.value = '';
      loginEmail.errorMessage = '';

      dispatch(setInputValueError(loginEmail));
    }

    if (loginPassword.value || loginPassword.errorMessage) {
      loginPassword.value = '';
      loginPassword.errorMessage = '';

      dispatch(setInputValueError(loginPassword));
    }

    return null;
  };
};

export default secondButtonCodePhone;
