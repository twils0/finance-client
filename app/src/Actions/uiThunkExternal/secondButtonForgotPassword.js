import { formNames, inputNames } from '../../Constants/uiConstantsExternal';
import { setCurrentForm, setInputValueError } from '../uiActionsExternal';

const secondButtonForgotPassword = () => async (dispatch, getState) => {
  const state = getState();
  const { forms } = state.ui.external;

  const forgotPasswordEmail =
    forms[formNames.FORGOT_PASSWORD].inputs[inputNames[formNames.FORGOT_PASSWORD].EMAIL];

  dispatch(setCurrentForm({ current: formNames.LOGIN }));

  if (forgotPasswordEmail.value || forgotPasswordEmail.errorMessage) {
    forgotPasswordEmail.value = '';
    forgotPasswordEmail.errorMessage = '';

    dispatch(setInputValueError(forgotPasswordEmail));
  }

  return null;
};

export default secondButtonForgotPassword;
