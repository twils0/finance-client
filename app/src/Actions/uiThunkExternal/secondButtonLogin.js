import { pathNames } from '../../Constants/universalConstants';
import { formNames, inputNames } from '../../Constants/uiConstantsExternal';
import { setInputValueError } from '../uiActionsExternal';

const secondButtonLogin = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'history')) {
    throw new Error(`Please enter a value for the 'history' key - ${JSON.stringify(payload)}`);
  }

  return async (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.external;

    const loginEmail = forms[formNames.LOGIN].inputs[inputNames[formNames.LOGIN].EMAIL];
    const loginPassword = forms[formNames.LOGIN].inputs[inputNames[formNames.LOGIN].PASSWORD];

    payload.history.replace(pathNames.SIGN_UP);

    if (loginEmail.value || loginEmail.errorMessage) {
      loginEmail.value = '';
      loginEmail.errorMessage = '';

      dispatch(setInputValueError(loginEmail));
    }

    if (loginPassword.value || loginPassword.errorMessage) {
      loginPassword.value = '';
      loginPassword.errorMessage = '';

      dispatch(setInputValueError(loginPassword));
    }

    return null;
  };
};

export default secondButtonLogin;
