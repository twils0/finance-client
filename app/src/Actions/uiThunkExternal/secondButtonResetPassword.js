import { requestStatusTypes } from '../../Constants/universalConstants';
import { statusNames } from '../../Constants/dataConstantsAuth';
import { formNames, inputNames } from '../../Constants/uiConstantsExternal';
import { setAuthStatus } from '../dataActionsAuth';
import { setCurrentForm, setInputValueError } from '../../Actions/uiActionsExternal';

const secondButtonResetPassword = () => async (dispatch, getState) => {
  const state = getState();
  const { forms } = state.ui.external;
  const { status } = state.data.auth;

  const forgotPasswordEmail =
    forms[formNames.FORGOT_PASSWORD].inputs[inputNames[formNames.FORGOT_PASSWORD].EMAIL];
  const resetPasswordCode =
    forms[formNames.RESET_PASSWORD].inputs[inputNames[formNames.RESET_PASSWORD].CODE];
  const resetPasswordPassword =
    forms[formNames.RESET_PASSWORD].inputs[inputNames[formNames.RESET_PASSWORD].PASSWORD];
  const resetPasswordPassword2 =
    forms[formNames.RESET_PASSWORD].inputs[inputNames[formNames.RESET_PASSWORD].PASSWORD2];

  if (forgotPasswordEmail.value || forgotPasswordEmail.errorMessage) {
    forgotPasswordEmail.value = '';
    forgotPasswordEmail.errorMessage = '';

    dispatch(setInputValueError(forgotPasswordEmail));
  }

  dispatch(setCurrentForm({ current: formNames.LOGIN }));

  if (status[statusNames.FORGOT_PASSWORD].status !== requestStatusTypes.IDLE) {
    dispatch(setAuthStatus({
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    }));
  }

  if (resetPasswordCode.value || resetPasswordCode.errorMessage) {
    resetPasswordCode.value = '';
    resetPasswordCode.errorMessage = '';

    dispatch(setInputValueError(resetPasswordCode));
  }

  if (resetPasswordPassword.value || resetPasswordPassword.errorMessage) {
    resetPasswordPassword.value = '';
    resetPasswordPassword.errorMessage = '';

    dispatch(setInputValueError(resetPasswordPassword));
  }

  if (resetPasswordPassword2.value || resetPasswordPassword2.errorMessage) {
    resetPasswordPassword2.value = '';
    resetPasswordPassword2.errorMessage = '';

    dispatch(setInputValueError(resetPasswordPassword2));
  }

  return null;
};

export default secondButtonResetPassword;
