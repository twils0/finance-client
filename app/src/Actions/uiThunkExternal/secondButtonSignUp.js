import { formNames, inputNames } from '../../Constants/uiConstantsExternal';
import { setCurrentForm, setInputValueError } from '../uiActionsExternal';

const secondButtonSignUp = () => async (dispatch, getState) => {
  const state = getState();
  const { forms } = state.ui.external;

  const signUpPassword = forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD];
  const signUpPassword2 = forms[formNames.SIGN_UP].inputs[inputNames[formNames.SIGN_UP].PASSWORD2];

  dispatch(setCurrentForm({ current: formNames.INFO }));

  if (signUpPassword.value || signUpPassword.errorMessage) {
    signUpPassword.value = '';
    signUpPassword.errorMessage = '';

    dispatch(setInputValueError(signUpPassword));
  }

  if (signUpPassword2.value || signUpPassword2.errorMessage) {
    signUpPassword2.value = '';
    signUpPassword2.errorMessage = '';

    dispatch(setInputValueError(signUpPassword2));
  }

  return null;
};

export default secondButtonSignUp;
