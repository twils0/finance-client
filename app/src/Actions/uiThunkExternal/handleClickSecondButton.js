import { formNames } from '../../Constants/uiConstantsExternal';
import { setCurrentForm } from '../uiActionsExternal';

import secondButtonResetPassword from './secondButtonResetPassword';
import secondButtonForgotPassword from './secondButtonForgotPassword';
import secondButtonLogin from './secondButtonLogin';
import secondButtonCard from './secondButtonCard';
import secondButtonSignUp from './secondButtonSignUp';
import secondButtonCodeMFAPhone from './secondButtonCodeMFAPhone';
import secondButtonCodeVerifyEmail from './secondButtonCodeVerifyEmail';
import secondButtonDevice from './secondButtonDevice';

const handleClickSecondButton = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'history')) {
    throw new Error(`Please enter a value for the 'history' key - ${JSON.stringify(payload)}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'resetStripeElement')) {
    throw new Error(
      `Please enter a value for the 'resetStripeElement' key - ${JSON.stringify(payload)}`,
    );
  }

  return (dispatch, getState) => {
    const state = getState();
    const { forms } = state.ui.external;

    switch (forms.current) {
      case formNames.RESET_PASSWORD:
        dispatch(secondButtonResetPassword());
        break;
      case formNames.FORGOT_PASSWORD:
        dispatch(secondButtonForgotPassword());
        break;
      case formNames.LOGIN: {
        const { history } = payload;

        dispatch(secondButtonLogin({ history }));
        break;
      }
      case formNames.CARD: {
        const { history, resetStripeElement } = payload;

        dispatch(secondButtonCard({ history, resetStripeElement }));
        break;
      }
      case formNames.INFO:
        dispatch(setCurrentForm({ current: formNames.CARD }));
        break;
      case formNames.SIGN_UP:
        dispatch(secondButtonSignUp());
        break;
      case formNames.CODE_MFA_PHONE: {
        const { history } = payload;

        dispatch(secondButtonCodeMFAPhone({ history }));
        break;
      }
      case formNames.CODE_VERIFY_EMAIL: {
        const { history } = payload;

        dispatch(secondButtonCodeVerifyEmail({ history }));
        break;
      }
      case formNames.DEVICE: {
        const { history } = payload;

        dispatch(secondButtonDevice({ history }));
        break;
      }
      default:
        break;
    }

    return null;
  };
};

export default handleClickSecondButton;
