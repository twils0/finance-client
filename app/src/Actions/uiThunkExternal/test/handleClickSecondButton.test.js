import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { initialState as initialStateUIExternal } from '../../../Reducers/uiReducersExternal';
import {
  actionTypes as actionTypesUIExternal,
  formNames,
} from '../../../Constants/uiConstantsExternal';
import handleClickSecondButton from '../handleClickSecondButton';
import secondButtonResetPassword from '../secondButtonResetPassword';
import secondButtonForgotPassword from '../secondButtonForgotPassword';
import secondButtonLogin from '../secondButtonLogin';
import secondButtonCard from '../secondButtonCard';
import secondButtonSignUp from '../secondButtonSignUp';
import secondButtonCodeMFAPhone from '../secondButtonCodeMFAPhone';
import secondButtonCodeVerifyEmail from '../secondButtonCodeVerifyEmail';
import secondButtonDevice from '../secondButtonDevice';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

jest.mock('../secondButtonResetPassword', () => jest.fn());
jest.mock('../secondButtonForgotPassword', () => jest.fn());
jest.mock('../secondButtonLogin', () => jest.fn());
jest.mock('../secondButtonCard', () => jest.fn());
jest.mock('../secondButtonSignUp', () => jest.fn());
jest.mock('../secondButtonCodeMFAPhone', () => jest.fn());
jest.mock('../secondButtonCodeVerifyEmail', () => jest.fn());
jest.mock('../secondButtonDevice', () => jest.fn());

const history = { test: 'testHistory' };
const handlerPayload = {
  history,
  resetStripeElement: jest.fn(),
};

describe('uiThunkExternal', () => {
  describe('handleClickSecondButton', () => {
    afterEach(() => {
      secondButtonResetPassword.mockReset();
      secondButtonForgotPassword.mockReset();
      secondButtonLogin.mockReset();
      secondButtonSignUp.mockReset();
      secondButtonCodeMFAPhone.mockReset();
      secondButtonCodeVerifyEmail.mockReset();
      secondButtonDevice.mockReset();
    });

    it("fails and throws an error when missing 'history' key in payload", async () => {
      const store = mockStore({
        ui: { external: initialStateUIExternal },
      });

      const emptyPayload = {};

      try {
        await store.dispatch(handleClickSecondButton(emptyPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }
    });

    it("fails and throws an error when missing 'resetStripeElement' key in payload", async () => {
      const store = mockStore({
        ui: { external: initialStateUIExternal },
      });

      const wrongPayload = { history };

      try {
        await store.dispatch(handleClickSecondButton(wrongPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }
    });

    it('calls nothing when forms current is wrong', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = 'testForm';

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
    });

    it('calls secondButtonResetPassword when form current is RESET_PASSWORD', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.RESET_PASSWORD;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonResetPassword.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonResetPassword).toBeCalled();
    });

    it('calls secondButtonForgotPassword when form current is FORGOT_PASSWORD', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.FORGOT_PASSWORD;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonForgotPassword.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonForgotPassword).toBeCalled();
    });

    it('calls secondButtonLogin when form current is LOGIN', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.LOGIN;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonLogin.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonLogin).toBeCalledWith({ history });
    });

    it('calls secondButtonCard when form current is CARD', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.CARD;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonCard.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonCard).toBeCalledWith(handlerPayload);
    });

    it('calls secondButtonLogin when form current is INFO', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.INFO;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      const expectedActions = [
        {
          type: actionTypesUIExternal.SET_CURRENT_FORM,
          payload: {
            current: formNames.CARD,
          },
        },
      ];

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
    });

    it('calls secondButtonSignUp when form current is SIGN_UP', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.SIGN_UP;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonSignUp.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonSignUp).toBeCalled();
    });

    it('calls secondButtonCodeMFAPhone when form current is CODE_MFA_PHONE', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.CODE_MFA_PHONE;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonCodeMFAPhone.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonCodeMFAPhone).toBeCalledWith({ history });
    });

    it('calls secondButtonCodeVerifyEmail when form current is CODE_VERIFY_EMAIL', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.CODE_VERIFY_EMAIL;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonCodeVerifyEmail.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonCodeVerifyEmail).toBeCalledWith({ history });
    });

    it('calls secondButtonDevice when form current is DEVICE', async () => {
      const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));
      stateBeforeUIExternal.forms.current = formNames.DEVICE;

      const store = mockStore({ ui: { external: stateBeforeUIExternal } });

      secondButtonDevice.mockReturnValue(() => null);

      const result = store.dispatch(handleClickSecondButton(handlerPayload));

      expect(result).toEqual(null);
      expect(secondButtonDevice).toBeCalledWith({ history });
    });
  });
});
