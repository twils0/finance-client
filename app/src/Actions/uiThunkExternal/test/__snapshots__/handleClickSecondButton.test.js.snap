// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiThunkExternal handleClickSecondButton fails and throws an error when missing 'history' key in payload 1`] = `[Error: Please enter a value for the 'history' key - {}]`;

exports[`uiThunkExternal handleClickSecondButton fails and throws an error when missing 'resetStripeElement' key in payload 1`] = `[Error: Please enter a value for the 'resetStripeElement' key - {"history":{"test":"testHistory"}}]`;
